scipy/stats/tests/test_multivariate.py

```diff
@@ -2479,3 +2481,107 @@ def test_random_state_property():
     for distfn, args in dists:
         check_random_state_property(distfn, args)
         check_pickling(distfn, args)
+
+class TestDirichletMultinomial:
+    def test_covariance_diagonal(self):
+        #Makes sure that the diagonal of the
+        #covariance matrix is the variance.
+        alpha = [1, 5, 3]
+        n = 3
+        c = dirichlet_multinomial.cov(alpha, n)
+        v = dirichlet_multinomial.var(alpha, n)
+
+        assert_array_equal(v, c.diagonal())
+
+    def test_variance(self):
+        alpha = [0.3, 2.1, 5.4]
+        n = 3
+        #Truncated to the 6th digit, so not
+        #exactly what we would expect from var
+        expected_v = [0.136162, 0.724381, 0.784293]
+        v = dirichlet_multinomial.var(alpha, n)
+        assert_array_almost_equal(expected_v, v)
+
+    def test_covariance(self):
+        alpha = [0.3, 2.1, 5.4]
+        n = 3
+        #Truncated to the 6th digit, so not
+        #exactly what we would expect from var
+        expected_c = np.array([[0.136162, -0.038125, -0.098037],
+                               [-0.038125, 0.724381, -0.686256],
+                               [-0.098037, -0.686256, 0.784293]])
+        c = dirichlet_multinomial.cov(alpha, n)
+        assert_array_almost_equal(expected_c, c)
+
+    def test_pmf(self):
+        x = np.array([1, 2, 3])
+        n = 3
+        alpha = np.array([3, 4, 5])
+        y = 2.884615384615385
+        y_1 = dirichlet_multinomial.pmf(alpha, x, n)
+        assert_almost_equal(y, y_1)
+
+    def test_logpmf(self):
+        x = np.array([1, 2, 3])
+        n = 3
+        alpha = np.array([3, 4, 5])
+        y = dirichlet_multinomial.pmf(alpha, x, n)
+        y_1 = dirichlet_multinomial.logpmf(alpha, x, n)
+        assert_almost_equal(y, np.exp(y_1))
+
+    def test_broadcasting(self):
```

Comment:
Combine what you can between this and the next test. Test all methods of the distribution, not just two. Ideally, strengthen the test using random lxmxn input rather than 2x2 input.